suc/update-connection-info.ps1

```diff
@@ -0,0 +1,220 @@
+    Write-LogInfo "Attempting to get updated Rancher connection info."
+    $env:RETRIEVED_CONNECTION_INFO = $false
+    while ($retries -lt 6) {
+        $responseCode = $(curl.exe --connect-timeout 60 --max-time 60 --write-out "%{http_code}\n " --ssl-no-revoke -sfL "$env:CATTLE_SERVER/v3/connect/agent" -o $env:CATTLE_AGENT_VAR_DIR/rancher2_connection_info.json -H "Authorization: Bearer $($env:CATTLE_TOKEN)" -H "X-Cattle-Id: $($env:CATTLE_ID)" -H "Content-Type: application/json")
+        switch ( $responseCode ) {
+            { $_ -in "ok200", 200 } {
+                Write-LogInfo "Successfully downloaded Rancher connection information."
+                $retries = 99
+                $env:RETRIEVED_CONNECTION_INFO = $true
+                break
+            }
+            default {
+                Write-LogError "$responseCode received while downloading Rancher connection information. Sleeping for 5 seconds and trying again."
+                Start-Sleep -Seconds 5
+                $retries++
+                continue
+            }
+        }
+    }
+    if ($env:RETRIEVED_CONNECTION_INFO -eq $false) {
+        Write-LogFatal "Could not retreieve connection information file from $env:CATTLE_SERVER."
+    }
+}
+
+function Update-WinsConfig()
+{
+    Write-LogDebug "Initial config file contents:"
+    Write-LogDebug $(Get-Content $env:CATTLE_AGENT_CONFIG_DIR/config | ForEach-object {Write-Host $_})
+
+    # Clear the old tls-config as it may no longer be applicable
+    Set-Content -Path $env:CATTLE_AGENT_CONFIG_DIR/config -Value (get-content -Path $env:CATTLE_AGENT_CONFIG_DIR/config | Select-String -SimpleMatch 'tls-config' -NotMatch)
+    Set-Content -Path $env:CATTLE_AGENT_CONFIG_DIR/config -Value (get-content -Path $env:CATTLE_AGENT_CONFIG_DIR/config | Select-String -SimpleMatch 'certFilePath' -NotMatch)
+
+    Write-LogDebug "Config file contents after tls-config removal:"
+    Write-LogDebug $(Get-Content $env:CATTLE_AGENT_CONFIG_DIR/config | ForEach-object {Write-Host $_})
+
+    if ((Test-Path -Path $env:RANCHER_CERT) -and ($env:CA_REQUIRED -eq "true"))
+    {
+        Write-LogInfo "Updating rancher-wins config file with updated TLS information."
+        # Update the tls-config with the new Rancher cert path
+        $tlsConfig =
+        @"
+tls-config:
+  certFilePath: $($( $env:RANCHER_CERT ).Replace("\\", "/") )
+"@
+        Add-Content -Path $env:CATTLE_AGENT_CONFIG_DIR/config -Value $tlsConfig
+    }
+
+    Write-LogDebug "Final config file contents:"
+    Write-LogDebug $(Get-Content $env:CATTLE_AGENT_CONFIG_DIR/config | ForEach-object {Write-Host $_})
+}
+
+function Update-ConnectionInfo()
+{
+    $env:RKE2_DATA_ROOT = "c:\var\lib\rancher"
+    $env:CATTLE_AGENT_CONFIG_DIR = "c:\etc\rancher\wins"
+
+    if ($env:CATTLE_WINS_CONFIG_DIR -and ($env:CATTLE_WINS_CONFIG_DIR -ne "")) {
+        $env:CATTLE_AGENT_CONFIG_DIR = $env:CATTLE_WINS_CONFIG_DIR
+    }
+
+    $env:CATTLE_AGENT_VAR_DIR = "$env:RKE2_DATA_ROOT\agent"
```

Comment:
```suggestion
    if ($env:CATTLE_AGENT_VAR_DIR -eq "") { 
      $env:CATTLE_AGENT_VAR_DIR = "$env:RKE2_DATA_ROOT\agent"
    }
```
I'd still probably do this here but it's up to you.

Reply:
On 2.9 there isn't a way for a user to configure this path during the provisioning process, so exposing it and allowing a user to change it via agent environment variables won't be helpful